### DISCUSSION TITLE
Fulltext search: How to display menu items in search results as well as pages

### DISCUSSION THREAD

**Discussion by FrancoisCamus** [OrchardCMS/OrchardCore#12725] [Q&A]
Hi guys! I'd like to be able to search menu items as well as pages. For example, after setting up a site with the blog recipe I can't find a way to display the menu items in the search results. I can get the main menu content item but not the actual children.

Any ideas how I could do this?

**Comment by Skrypt** [comment]
I would need to try but I think it could work by using the Full-Text index settings on the Menu content type. You could try to do a Liquid `foreach menuitems in menu` in there.

**Reply by FrancoisCamus** [reply]
I've added {{ Model.Content.MenuItemsListPart | full_text_aspect }} as custom full-text for Menu content type. But when I do a search it is bringing up the actual main menu and not the menu items of this main menu.

**Reply by Skrypt** [reply]
You need to do something like:

```liquid
{% assign MenuItems = Model.Content.MenuItemsListPart | content_item_id %}
{% for MenuItem in MenuItems %}
  {{ MenuItem | display_text }}
{% endfor %}
```

**Reply by FrancoisCamus** [reply]
Thanks for this. I've got the same behavior. When I search for a menu item I get a unique result which is the main menu content item. It is matching the text but giving me the parent as a result not the child menu item.

**Reply by FrancoisCamus** [reply]
Oups sorry I forgot to rebuild my index. It doesn't work with the code you provided. Would this code actually help to list the menu items as results?

**Comment by FrancoisCamus** [comment]
It seems that menu items are not content items. They are created within a menu and stored with it. As it is, I cannot query them with a lucene query. I can only query for the parent menu which is not what I want.

Not sure what the best way should be to implement this. I could modfy the SearchController to return the menu items whenever I get a menu and create a new template. Or I could try to modify the menu so it uses a list part or something like that so then I can make the menu items creatable and listable that way I could search for them. But that would break the way menu items are stored at the moment.

Any suggestions?

**Comment by FrancoisCamus** [comment]
Is it possible to create a Lucene query that would return menu items? I can get content items of type Menu but not menu items.

**Reply by Skrypt** [reply]
You will need to create a custom indexing provider for handling menu items. Else we can add also this to the source code as an enhancement.

**Reply by FrancoisCamus** [reply]
Alright thanks! Where would you suggest me to look as a starting point? I don't know how this is implemented in Orchard Core.
